### DISCUSSION TITLE
Exanding injections without removing upstream values

### DISCUSSION THREAD

**Discussion by traxys** [nvim-treesitter/nvim-treesitter#4690] [Q&A] [answered]
Is it possible in some way to add additional injections without overriding the upstream injections ? 

In my case I want to add an injection to `nix` files that a value of the form `extraConfigLua = "<some string>";` is highlighted as lua code.
I did this by the following:

```lua
local luaInjections = [[
        (binding
          attrpath: (attrpath (identifier) @_path)
          expression: [
            (string_expression (string_fragment) @lua)
            (indented_string_expression (string_fragment) @lua)
          ]
          (#match? @_path "^extraConfigLua(Pre|Post)?$"))
]]

vim.treesitter.query.set("nix", "injections", luaInjections)
```

But this will override the queries that allow `bash` highlighting in other `nix` strings. Is there a way to make this work ?

**Comment by clason** [comment, marked as answer]
Yes, add it to a query file in runtimepath that starts with `; extends`, see `:h treesitter-query-modeline-extends`.

**Reply by traxys** [reply]
Thank you! I searched for a bit in around the help vim.treesitter.query.set but did not encounter that one. Is there no programatic way (it would be simpler in my case, but I should be able to make the query file work) ?

**Reply by clason** [reply]
I don't think so, no. You can only get the parsed query (`:h vim.treesitter.query.get()`), but that is not in a format you can manipulate in this way.
